js/modulestest/runtime.go

```diff
@@ -0,0 +1,51 @@
+// Package modulestest contains helpers to test js modules
+package modulestest
+
+import (
+	"context"
+	"testing"
+
+	"github.com/dop251/goja"
+	"go.k6.io/k6/js/common"
+	"go.k6.io/k6/js/eventloop"
+	"go.k6.io/k6/lib"
+)
+
+// Runtime is a helper struct that contains what is needed to run a (simple) module test
+type Runtime struct {
+	VU            *VU
+	EventLoop     *eventloop.EventLoop
+	CancelContext func()
+}
+
+// NewRuntime will create a new test runtime and will cancel the context on test/benchmark end
+func NewRuntime(t testing.TB) *Runtime {
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+	vu := &VU{
+		CtxField:     ctx,
+		RuntimeField: goja.New(),
+	}
+	vu.RuntimeField.SetFieldNameMapper(common.FieldNameMapper{})
+	eventloop := eventloop.New(vu)
+	vu.RegisterCallbackField = eventloop.RegisterCallback
+	result := &Runtime{
+		VU:            vu,
+		EventLoop:     eventloop,
+		CancelContext: cancel,
+	}
+	// let's cancel again in case it has changed
+	t.Cleanup(func() { result.CancelContext() })
+	return result
+}
+
+// InitContext will set the InitContext of the Runtime
+func (r *Runtime) InitContext(i *common.InitEnvironment) {
+	r.VU.InitEnvField = i
+}
+
+// MoveToVUContext will set the state and nil the InitEnv just as a real VU
+func (r *Runtime) MoveToVUContext(state *lib.State) {
+	r.VU.InitEnvField = nil
+	r.VU.StateField = state
```

Comment:
hmm the `Field` suffix of all of these struct fields kind of annoys me now, but it's a minor nitpick :sweat_smile:

Reply:
I have to give them name and I can't remove the `Field` as that is the name of the method defined on the interface ;0

Reply:
Also, at least in theory this should only be used when defining the test, and then you should just use `vu.State()` to get back the `lib.State`. And in the majority of the test it seems like you barely need to touch those after figuring out some default test setup for the particular package tests.

Reply:
yeah, I know, but still unreasonably annoyed by them, and I don't have any better suggestions...  :sweat_smile: